modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/typeclasses/suspended/FxSyntax.kt

```diff
@@ -40,39 +36,46 @@ interface FxSyntax<F> : Concurrent<F>, BindSyntax<F> {
   fun <A> CoroutineContext.parSequence(effects: Iterable<Kind<F, A>>): Kind<F, List<A>> =
     parTraverse(effects, ::identity)
 
-  fun <A> effect(fa: suspend () -> A): Kind<F, A> =
-    async { cb ->
-      fa.startCoroutine(asyncContinuation(EmptyCoroutineContext, cb))
-    }
-
   fun <A> ensure(fa: suspend () -> A, error: () -> Throwable, predicate: (A) -> Boolean): Kind<F, A> =
     run<Monad<F>, Kind<F, A>> { fa.effect().ensure(error, predicate) }
 
-  private fun <A> asyncOp(fb: Async<F>.() -> Kind<F, A>): Kind<F, A> =
-    run<Async<F>, Kind<F, A>> { fb(this) }
-
-  fun <A> CoroutineContext.effect(f: suspend () -> A): Kind<F, A> =
-    asyncOp { defer(this@effect) { f.effect() } }
-
-  fun <A> (suspend () -> A).effect(unit: Unit = Unit): Kind<F, A> = effect(this)
+  fun <A> (suspend () -> A).effect(unit: Unit = Unit): Kind<F, A> = effect(f = this)
 
-  fun <A, B> (suspend (A) -> B).effect(): (Kind<F, A>) -> Kind<F, B> =
-    { suspend { this(it.bind()) }.effect() }
-
-  fun <A, B, C> (suspend (A, B) -> C).effect(): (Kind<F, A>, Kind<F, B>) -> Kind<F, C> =
-    { ka, kb -> suspend { this(ka.bind(), kb.bind()) }.effect() }
+  fun <A, B> (suspend (A) -> B).effect(): (Kind<F, A>) -> Kind<F, B> = { fa ->
+    fa.flatMap { a ->
+      effect { this(a) }
+    }
+  }
 
-  fun <A, B, C, D> (suspend (A, B, C) -> D).effect(): (Kind<F, A>, Kind<F, B>, Kind<F, C>) -> Kind<F, D> =
-    { ka, kb, kc -> suspend { this(ka.bind(), kb.bind(), kc.bind()) }.effect() }
+  fun <A, B, C> (suspend (A, B) -> C).effect(): (Kind<F, A>, Kind<F, B>) -> Kind<F, C> = { ka, kb ->
```

Comment:
We need to check if we need these as they may be too esoteric for the public api